### PR TITLE
Some objects in the OpenAPI spec are incorrect

### PR DESCRIPTION
**Pull request by jkoenig134**
# Readiness checklist

-   [ ] I added/updated tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description

Fixes nmshd/feedback#29

Fixed issues:
- `@type` is displayed as required in `RequestResponseContent`
- `peerIdentity` is missing in `Relationship`

### CODE REVIEW

**Comment by jkoenig134** [comment]
I don't get your question. In a perfect world openapi and the results would be 1:1 the same. It's hard to archieve that, so I fixed these bugs.
If you find more feel free to mention that :)

File: src/modules/coreHttpApi/openapi.yml [hunk 1]
```diff
@@ -5615,6 +5614,20 @@ components:
             - $ref: "#/components/schemas/Address"
           nullable: false
           description: The address of the peer identity.
+        peerIdentity:
```
Comment (Milena-Czierlinski) [inline src/modules/coreHttpApi/openapi.yml:5617]: `peerIdentity` must be added below to required. Also, to be more consistent with our code, I'd prefer it if you move this below `peerDeletionInfo`.

Comment (jkoenig134) [reply src/modules/coreHttpApi/openapi.yml:5617]: It's consistent to how the object will occur so I leave it where it is.

**Comment by Milena-Czierlinski** [comment]
I had trouble finding the data structure that is returned as a result so that I might compare them. Where do I see how the returned object will occur?

Another question that is principally unrelated to your changes: Why do we have `peer` and `peerIdentity`?

**Comment by jkoenig134** [comment]
> I had trouble finding the data structure that is returned as a result so that I might compare them. Where do I see how the returned object will occur?
> 
> Another question that is principally unrelated to your changes: Why do we have `peer` and `peerIdentity`?

The Data structure is RelationshipDTO.

Regarding `peer` and `peerIdentity`: I think for easier / more performant querying.

**Comment by Milena-Czierlinski** [comment]
And is `@type` never required?

**Comment by jkoenig134** [comment]
`@type` is not rendered out by ts-serval when the property can only have one `@type`, to save bandwidth.

**Comment by britsta** [comment]
> Regarding `peer` and `peerIdentity`: I think for easier / more performant querying.

Should we also add the `peerIdentity` property to the documentation of the Relationship in the Data Model Overview then? I also thought it was weird to have a `peer` property in addition to the `peerIdentity` property. I wonder how we can present this redundance in a meaningful way in the documentation.

**Comment by jkoenig134** [comment]
we should document it, yes.

**Review by britsta — APPROVED** [review]

**Pull request merged**